frontend/src/main/web/src/components/OverlayTrigger/index.jsx

```diff
@@ -0,0 +1,294 @@
+import React, { cloneElement, Component, PropTypes } from 'react'
+import ReactDOM from 'react-dom'
+import contains from 'dom-helpers/query/contains'
+import warning from 'warning'
+import { pick } from 'lodash'
+import createChainedFunction from '../../utils/createChainedFunction'
+import Overlay from '../Overlay'
+/**
+ * Check if value one is inside or equal to the of value
+ *
+ * @param {string} one
+ * @param {string|array} of
+ * @returns {boolean}
+ */
+function isOneOf (one, of) {
+  if (Array.isArray(of)) {
+    return of.indexOf(one) >= 0
+  }
+  return one === of
+}
+
+class OverlayTrigger extends Component {
+  state = {
+    isOverlayShown: this.props.defaultOverlayShown
+  }
+  show = () => {
+    this.setState({
+      isOverlayShown: true
+    })
+  }
+  hide = () => {
+    this.setState({
+      isOverlayShown: false
+    })
+  }
+  toggle = () => {
+    if (this.state.isOverlayShown) {
+      this.hide()
+    } else {
+      this.show()
+    }
+  }
+  renderOverlay = () => {
+    ReactDOM.unstable_renderSubtreeIntoContainer(
+      this, this._overlay, this._mountNode
+    )
+  }
+  getOverlayTarget = () => {
+    return ReactDOM.findDOMNode(this)
+  }
+  getOverlay = () => {
+    let overlayProps = {
+      ...pick(this.props, Object.keys(Overlay.propTypes)),
+      show: this.state.isOverlayShown,
+      placement: this.props.overlay.props.placement || this.props.placement,
+      onHide: this.hide,
+      target: this.getOverlayTarget,
+      onExit: this.props.onExit,
+      onExiting: this.props.onExiting,
+      onExited: this.props.onExited,
+      onEnter: this.props.onEnter,
+      onEntering: this.props.onEntering,
+      onEntered: this.props.onEntered
+    }
+
+    let overlay = cloneElement(this.props.overlay, {
+      placement: overlayProps.placement,
+      container: overlayProps.container
+    })
+
+    return (
+      <Overlay {...overlayProps}>
+        { overlay }
+      </Overlay>
+    )
+  }
+  handleDelayedShow = () => {
+    if (this._hoverHideDelay != null) {
```

Comment:
This looks like it is checking that no timeout is set, but the initial value will be `undefined` not `null`. It looks like a coercing comparison operator (`!=`) is being used to avoid the issue with this, but it would be better just to use `undefined` in all cases here rather than `null` and use the usual non-coercing operator `!==` to avoid any possible issues with future maintenance.

It also looks like the name is misleading. This thing is a timeout handle, but the name suggests that it is an amount of time to delay it. Can we call it something like `hoverHideTimeoutHandle` to make that clear?